lib/liquid/comment_body.rb

```diff
@@ -0,0 +1,36 @@
+module Liquid
+  class CommentBody < BlockBody
+    def parse(tokenizer, parse_context)
+      parse_context.line_number = tokenizer.line_number
+      while token = tokenizer.shift
+        unless token.empty?
+          if token.start_with?(TAGSTART)
+            whitespace_handler(token, parse_context)
+            if token =~ FullToken
+              tag_name = $1
+              markup = $2
+
+              if tag_name == 'comment'
+                new_tag = Comment.parse(tag_name, markup, tokenizer, parse_context)
+                @blank &&= new_tag.blank?
+                @nodelist << new_tag
+              else
```

Comment:
I think you need to handle the `raw` tag as well, given that it was previously the method used to workaround syntax errors in comments.